department-of-veterans-affairs/vets-api
# [39784] Updating UserVerified to throw an error if user is previously…

… verified and a new login shows an ICN that already exists on another user

## Description of change
This PR refactors UserVerifier to make it easier to understand the code flow, as well as adding the functionality of throwing an error if a user has an existing UserVerification with UserAccount that is verified, and the incoming authenticated user has  a different ICN that matches to a separate UserAccount. This can only occur if a user has previously verified, received an ICN, and then had their ICN changed to a user we consider verified, which is a situation we should probably block and force manual intervention to resolve

## Original issue(s)
department-of-veterans-affairs/va.gov-team#39784

## Things to know about this PR
- To test, find a test user and get their ICN
- Create a UserAccount -> `UserAccount.new(icn: <test-user-icn>).save!`
- Create a UserVerification that associates with the UserAccount -> `UserVerification.new(idme_uuid: 'some-idme-uuid', user_account: <created-user-account-above>).save!`
- This represents an existing user that is wholly separate from our test user, yet somehow has their ICN
- Now create a UserAccount/UserVerification to represent the 'previous state' of the test user, you'll need to get the `idme_uuid` of the test user
- `UserAccount.new(icn: 'some-random-icn').save!`
- `UserVerification.new(idme_uuid: <test-user-idme-uuid>).save!`
- Now attempt to log in with the test user, the UserVerification service object will attempt to reconcile the fact that the test user should be attached to the UserAccount initially created, but we'll be in a bad state, so auth should be blocked
